### PR TITLE
Feed migrations to support multi-domain

### PR DESCRIPTION
**Pull request by sonika-shah**
### Describe your changes:

Fixes  
- [x] domain related feeds not generated after multi-domain changes
- [x]  domain still null in thread_entity table after adding column
- [x] filter threads by domain 
- [x] Tests

#
### Type of change:

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/test/java/org/openmetadata/service/resources/feeds/FeedResourceTest.java [hunk 1]
```diff
@@ -1486,6 +1504,504 @@ void patch_post_reactions_200() throws IOException {
...
+      Table tableDomain1 = createTableWithDomain(testDomain.getEntityReference());
+      Table tableNoDomain = createTableWithDomain(null);
+
+      try {
+        // Create threads
+        Thread threadDomain1 =
+            createAndCheck(
+                create()
+                    .withAbout(buildEntityLink(Entity.TABLE, tableDomain1.getFullyQualifiedName()))
+                    .withDomains(List.of(tableDomain1.getId())),
```
[the remark this answers is not in the document]
Comment (sonika-shah) [reply openmetadata-service/src/test/java/org/openmetadata/service/resources/feeds/FeedResourceTest.java:1829]: Table tableDomain1 = createTableWithDomain(testDomain.getEntityReference());
  table has the same domain as specified

**Comment by mohityadav766** [comment]
@claude can you review the Pr!

**Comment by sonika-shah** [comment]
-- MySQL - will create ["null"] when domain is literally "null" 
JSON_ARRAY(IF(JSON_EXTRACT(...) IS NOT NULL, JSON_EXTRACT(...), NULL))

-- PostgreSQL - will create [null] when domain is null
CASE WHEN json#>'{...domain}' IS NULL THEN ARRAY[]::jsonb[] ELSE ARRAY[json#>'{...domain}'] END

we are never storing 

> **Claude finished @mohityadav766's task** —— [View job](https://github.com/open-metadata/OpenMetadata/actions/runs/16681013936)
> 
> ### PR Review: Feed migrations to support multi-domain
> This PR successfully migrates the feed system from single domain support to multi-domain arrays. I've reviewed the changes and found them to be well-implemented overall with good test coverage.
> 
> ## ✅ **Strengths**
> ### **Database Migration**
> * **Comprehensive entity coverage**: All 30+ entity tables properly migrated from `domain` to `domains`
> * **Database-specific implementations**: Proper MySQL vs PostgreSQL syntax handling
> * **Thread-specific migration**: Both root-level and nested `feedInfo.entitySpecificInfo.entity.domain` paths handled
> 
> ### **Schema Updates**
> * **Consistent data model**: `thread.json` and `changeEvent.json` properly updated to use UUID arrays
> * **Backward compatibility**: Generated TypeScript types updated automatically
> 
> ### **Service Layer**
> * **Smart domain filtering**: `FeedFilter.java:67-97` implements database-specific JSON array queries
> * **Clean UUID conversion**: `FeedUtils.java:148-151` properly converts EntityReference to UUID arrays
> * **ChangeEvent support**: `FormatterUtil.java:302-305,373` handles domain propagation in events
> 
> ### **Test Coverage**
> * **Edge case testing**: Users with no domains, mixed domain access scenarios
> * **Multi-domain scenarios**: Different thread types with domain inheritance
> * **574 lines of comprehensive test coverage**
> 
> ## ⚠️ **Issues to Address**
> ### **1. Critical: Null Array Creation in Migrations**
> **Lines**: `bootstrap/sql/migrations/native/1.9.0/mysql/schemaChanges.sql:164-169`, `postgres/schemaChanges.sql:197-200`
> 
> The migrations can create arrays containing `null` values when `JSON_EXTRACT`/`json#>` returns null, despite WHERE conditions:
> 
> ```sql
> -- MySQL - will create ["null"] when domain is literally "null" 
> JSON_ARRAY(IF(JSON_EXTRACT(...) IS NOT NULL, JSON_EXTRACT(...), NULL))
> 
> -- PostgreSQL - will create [null] when domain is null
> CASE WHEN json#>'{...domain}' IS NULL THEN ARRAY[]::jsonb[] ELSE ARRAY[json#>'{...domain}'] END
> ```
> 
> **Fix**: Use explicit null checks in CASE statements as suggested by existing review comments.
> 
> ### **2. Test Logic Issue**
> **Line**: `FeedResourceTest.java:1829`
> 
> ```java
> .withDomains(List.of(tableDomain1.getId())), // Should be testDomain.getId()
> ```
> 
> The test creates a table with `testDomain` but then creates a thread with `tableDomain1.getId()` (table's ID instead of domain's ID).
> 
> ### **3. Missing Error Handling**
> * No validation that migrated domain arrays contain valid UUIDs
> * No rollback strategy if migrations partially fail
> * FeedFilter database queries could fail silently with malformed JSON
> 
> ## 🔧 **Minor Improvements**
> ### **Performance Optimization**
> * Consider adding indexes on the new `domains` generated column for faster queries
> * The JSON array operations in FeedFilter could benefit from caching for frequently accessed domains
> 
> ### **Code Quality**
> * `FeedFilter.java:68-82`: Complex JSON_TABLE query could be extracted to a helper method
> * ActivityFeedEvents.json: The field change from `domain` to `domains` is correct but consider version compatibility
> 
> ## 📋 **Verification Checklist**
> * ✅ All entity tables include domain migration
> * ✅ Both MySQL and PostgreSQL syntax handled
> * ✅ Thread filtering works with multi-domain arrays
> * ✅ Test coverage for edge cases
> * ⚠️ Null handling in migrations needs fixing
> * ⚠️ Test assertion logic needs correction
> 
> ## 🎯 **Recommendation**
> **Approve with fixes required**: The core implementation is solid and addresses the multi-domain migration comprehensively. The null handling issues in migrations should be fixed before merging to prevent data corruption.
> 
> #### Todo List:
> * [x]  Analyze database migration scripts (MySQL/PostgreSQL)
> * [x]  Review schema changes for thread/changeEvent entities
> * [x]  Review Java service layer changes (FeedFilter, FeedUtils, etc.)
> * [x]  Review test coverage and implementation
> * [x]  Assess null handling and edge cases
> * [x]  Provide comprehensive review feedback

we are never storing domain as "nul" it was always null or an UUID, so we are good here 
For the test concern, the table belongs to testDomain, so the actual check is correct
